src/files-breadcrumbs.tsx

```diff
@@ -49,19 +49,47 @@ function useHostname() {
 
 // eslint-disable-next-line max-len
 export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: string[], showHidden: boolean, setShowHidden: React.Dispatch<React.SetStateAction<boolean>>}) {
+    const [editMode, setEditMode] = React.useState(false);
+    const [newPath, setNewPath] = React.useState(path.join("/"));
     const hostname = useHostname();
 
     function navigate(n_parts: number) {
         cockpit.location.go("/", { path: encodeURIComponent(path.slice(0, n_parts).join("/")) });
     }
 
+    const handleInputKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === "Enter") {
+            changePath();
+        } else if (event.key === "Escape") {
+            cancelPathEdit();
+        }
+    };
+
+    const changePath = () => {
+        setEditMode(false);
+        // HACK: strip trailing / to circumvent the path being `//` in breadcrumbs
+        cockpit.location.go("/", { path: encodeURIComponent(newPath.replace(/\/$/, '')) });
```

Comment:
Ugh -- that's a page reload, right? Does every path change incur a page reload? That cries for `path` being a top-level state (we even have contexts now, don't we?), and adding a `setPath` callback property to  FilesBreadcrumbs which can change the global state.

Reply:
afaik, it only changes the hash `            window.location.hash = '#' + href;` to add the path there, it does cause a re-render. But we need to call cockpit.location.go to add it to history.